Arduino Uno Q compile issue

A quick heads up so that everyone who is diligent enough to look at the issues can see that Team RadioLib know about this issue.

It occurs both in the standard Arduino 2.x IDE and in App Lab. I'll check PlatformIO later.

### Compile output

```
/Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.cpp: In member function 'virtual void ArduinoHal::pinMode(uint32_t, uint32_t)':
/Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.cpp:25:52: error: invalid conversion from 'uint32_t' {aka 'unsigned int'} to 'PinMode' [-fpermissive]
   25 |   ::pinMode(pin, RADIOLIB_ARDUINOHAL_PIN_MODE_CAST mode);
      |                                                    ^~~~
      |                                                    |
      |                                                    uint32_t {aka unsigned int}
In file included from /Volumes/AppSupport/Arduino15/packages/arduino/hardware/zephyr/0.51.0/cores/arduino/api/Interrupts.h:27,
                 from /Volumes/AppSupport/Arduino15/packages/arduino/hardware/zephyr/0.51.0/cores/arduino/api/ArduinoAPI.h:29,
                 from /Volumes/AppSupport/Arduino15/packages/arduino/hardware/zephyr/0.51.0/cores/arduino/Arduino.h:9,
                 from /Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/../../BuildOpt.h:127,
                 from /Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/../../TypeDef.h:6,
                 from /Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.h:2,
                 from /Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.cpp:1:
/Volumes/AppSupport/Arduino15/packages/arduino/hardware/zephyr/0.51.0/cores/arduino/api/Common.h:115:44: note:   initializing argument 2 of 'void pinMode(pin_size_t, PinMode)'
  115 | void pinMode(pin_size_t pinNumber, PinMode pinMode);
      |                                    ~~~~~~~~^~~~~~~
/Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.cpp: In member function 'virtual void ArduinoHal::digitalWrite(uint32_t, uint32_t)':
/Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.cpp:32:59: error: invalid conversion from 'uint32_t' {aka 'unsigned int'} to 'PinStatus' [-fpermissive]
   32 |   ::digitalWrite(pin, RADIOLIB_ARDUINOHAL_PIN_STATUS_CAST value);
      |                                                           ^~~~~
      |                                                           |
      |                                                           uint32_t {aka unsigned int}
/Volumes/AppSupport/Arduino15/packages/arduino/hardware/zephyr/0.51.0/cores/arduino/api/Common.h:116:51: note:   initializing argument 2 of 'void digitalWrite(pin_size_t, PinStatus)'
  116 | void digitalWrite(pin_size_t pinNumber, PinStatus status);
      |                                         ~~~~~~~~~~^~~~~~
/Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.cpp: In member function 'virtual void ArduinoHal::attachInterrupt(uint32_t, void (*)(), uint32_t)':
/Users/nick/Documents/Arduino/libraries/RadioLib/src/hal/Arduino/ArduinoHal.cpp:46:89: error: invalid conversion from 'uint32_t' {aka 'unsigned int'} to 'PinStatus' [-fpermissive]
   46 |   ::attachInterrupt(interruptNum, interruptCb,  RADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST mode);
      |                                                                                         ^~~~
      |                                                                                         |
      |                                                                                         uint32_t {aka unsigned int}
/Volumes/AppSupport/Arduino15/packages/arduino/hardware/zephyr/0.51.0/cores/arduino/api/Common.h:132:82: note:   initializing argument 3 of 'void attachInterrupt(pin_size_t, voidFuncPtr, PinStatus)'
  132 | void attachInterrupt(pin_size_t interruptNumber, voidFuncPtr callback, PinStatus mode);
      |                                                                        ~~~~~~~~~~^~~~

```

This is using the LW Starter app with some keys.


**Additional info:**
 - MCU: Arduino UNO Q 
 - Link to Arduino core: Probably https://github.com/arduino/ArduinoCore-zephyr - which means we now have an abstraction level (Arduino) running on top of the Zephyr abstraction level - and Zephyr takes abstraction to a whole new level of opaque
 - Wireless module type NA
 - Arduino IDE version: 2.3.6 and App Lab 0.2.0
 - Library version: 7.4.0

### Things I've tripped up on:

Serial no longer comes out of a USB lead - you have to add a USB to Serial lead to the D0/D1 pins if you want Serial to work, but not tried it. Monitor is the replacement in App Lab but that doesn't appear to be working.

I've done a quick low pass on the sofa last night on this board, it has some merit for AI at the edge, for @StevenCellist's benefit, it seems to need about 140mA at rest and peaks at 250mA.

--- a/src/BuildOpt.h
+++ b/src/BuildOpt.h
@@ -388,6 +388,13 @@
   #define RADIOLIB_ARDUINOHAL_PIN_STATUS_CAST         (PinStatus)
   #define RADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST     (PinStatus)
 
+#elif defined(ARDUINO_ARCH_ZEPHYR) && defined(ARDUINO_UNO_Q)
+  // Arduino Uno Q (Zephyr OS)
+  #define RADIOLIB_PLATFORM                           "Arduino Uno Q (Zephyr OS)"
+  #define RADIOLIB_ARDUINOHAL_PIN_MODE_CAST           (PinMode)
+  #define RADIOLIB_ARDUINOHAL_PIN_STATUS_CAST         (PinStatus)
+  #define RADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST     (PinStatus)
+
 #else
   // other Arduino platforms not covered by the above list - this may or may not work
   #define RADIOLIB_PLATFORM                           "Unknown Arduino"

